prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -75,23 +82,90 @@ abstract class Event(val eventAction: EventAction, val at: OffsetDateTime?) {
     companion object {
         fun parseQueueMessage(event: String): Event {
             val parts = event.split(messageDelimiter)
-            if (parts.size < 3 || parts.size > 4) error("Internal Error: bad event format")
+            if (parts.size < 3 || parts.size > 7) error("Internal Error: bad event format")
```

Comment:
Can you explain why the 7?  Maybe even add a comment here.  From the code just below, maybe this error can be incorporated into the when based on type?  I know this would be an improvement on old code.  Maybe consider it.

Reply:
Pulled our splitting and verification into separate private function, included parts length checking on a per-event-type basis.